deploy/twingate-operator/crds/twingate.com.twingateresourceaccesses.yaml

```diff
@@ -16,15 +16,41 @@ spec:
             spec:
               type: object
               description: "TwingateResourceAccessSpec defines the desired state of TwingateResourceAccess"
-              x-kubernetes-validations:
-                - rule: self.principalId == oldSelf.principalId
-                  message: "principalId is immutable"
+              oneOf:
+                - properties:
+                  required: ["resourceRef", "principalId"]
+                - properties:
+                  required: ["resourceRef", "principalExternalRef"]
```

Comment:
Does this prevent us from specifying both `principalId` and `principalExternalRef`? 🤔

Reply:
yes.... it says `oneOf`

Reply:
yes.
also added test